src/k8s-engine/images/imageProcessor.ts

```diff
@@ -504,6 +504,81 @@ export abstract class ImageProcessor extends EventEmitter {
 
     console.log(`Installing kim: kim ${ args.join(' ') }`);
 
+    // Need two loops here. First loop is to verify that we an successfully run `kim builder install`.
+    // If it fails because kubernetes is still running a docker node, we wait and retry, assuming that
+    // eventually it will stop using the docker node and will be running the newer containerd-based node.
+    // You can run `watch kubectl get node -o 'jsonpath={.items[].status.nodeInfo.containerRuntimeVersion}'`
+    // to see how this changes.
+    //
+    // If `kim builder install` fails for any other reason,
+    // this is treated as a fatal but non-terminating error -- the user is advised that they might
+    // need to reset kubernetes in order to build images with nerdctl.
+    //
+    // Once it can successfully run `docker builder install`, it moves to the next loop.
+    // Otherwise it fails on a timeout, again with a logged suggestion to reset kubernetes.
+    while (true) {
+      try {
+        await childProcess.spawnFile(
+          resources.executable('kim'),
+          args,
+          {
+            stdio:       ['ignore', 'pipe', 'pipe'],
+            windowsHide: true,
+          });
+        break;
+      } catch (e) {
+        console.error(`Failed to restart the kim builder: ${ e.message }.`);
+        if (!e.stderr?.includes('Error: container runtime `docker` not supported')) {
+          console.error(`Attempt to run 'kim install builder' => error: '${ e.stderr }'`);
+          console.error('A reset might be necessary to support building images.');
+
+          return;
+        }
+      }
+
+      console.log('Waiting for the kubernetes backend to start using the newest nodes.');
+      if ((Date.now() - startTime) > maxWaitTime) {
+        console.log(`Waited more than ${ maxWaitTime / 1000 } secs; Probably a reset is needed to build images.`);
+        break;
+      }
+      await util.promisify(setTimeout)(waitTime);
+    }
+
+    // And now wait for the builder to be ready
+    while (true) {
+      if (await backend.isServiceReady('kube-image', 'builder')) {
+        console.log('Image-building support is now installed and running');
```

Comment:
```suggestion
        console.debug('Image-building support is now installed and running');
```
This isn't worth logging in the normal case.

Reply:
done

Reply:
Actually, after dealing with the temporary problems when switching from `moby` to `containerd`, I found the logs looked better in non-debug to have a statement that things are working now after the repeated error message when kubernetes is still using a moby-based node.